plugins/kubernetes-backend/src/cluster-locator/index.ts

```diff
@@ -18,30 +18,45 @@ import { Config } from '@backstage/config';
 import { ClusterDetails } from '../types/types';
 import { ConfigClusterLocator } from './ConfigClusterLocator';
 import { GkeClusterLocator } from './GkeClusterLocator';
+import { Logger } from 'winston';
 
 export const getCombinedClusterDetails = async (
   rootConfig: Config,
+  logger: Logger,
 ): Promise<ClusterDetails[]> => {
+  const clusterLocatorMethodsConfig = rootConfig.getOptionalConfigArray(
+    'kubernetes.clusterLocatorMethods',
+  );
+
+  if (!clusterLocatorMethodsConfig) {
```

Comment:
This seems to be an ok way to handle missing config in this function. The problem is there are several other functions that try to get k8s config, and throw.

Reply:
Yep gotta say this does seem a bit tedious, and easy to break in the future too. Do you think a simple top-level `config.has('kubernetes')` at the plugin entry poinit with an early return might be just as effective?

Reply:
Ah realized that's your other option. I'd say just flip that one around so that you're checking whether config is missing and do an early return in that case, or throw depending on if we're in dev. You can probably return `{ router: Router() }` to make it work alright

Reply:
@Rugvip ok, applied your feedback!

Do you think this needs tests or docs or anything else?